src/shared/reboot-util.c

```diff
@@ -107,3 +108,43 @@ int shall_restore_state(void) {
 
         return r > 0 ? ret : true;
 }
+
+static int xen_kexec_loaded(void) {
+#if HAVE_XENCTRL
+        xc_interface *xch;
+        int r;
+
+        r = dlopen_xenctrl();
+        if (r < 0)
+                return r;
+
+        if (access("/proc/xen", F_OK) < 0)
+                return log_debug_errno(errno, "Cannot access /proc/xen: %m");
+
+        xch = sym_xc_interface_open(NULL, NULL, 0);
+        if (!xch)
+                return log_debug_errno(errno, "Failed to open Xen control interface: %m");
+
+        r = sym_xc_kexec_status(xch, KEXEC_TYPE_DEFAULT);
```

Comment:
hmm, what does this return? I presume that if also returns errors in errno, no? 

hence, shouldn't this be:

```c
r = RET_NERRNO(sym_xc_kexec_status(xch, KEXEC_TYPE_DEFAULT));
```

(under the assumtpion that these calls communicate errors via errno, as your xc_interface_open() call above appears to suggests)

Reply:
> hmm, what does this return? I presume that if also returns errors in errno, no?

No, in this case it returns the error as negative value.